ecs-agent/managed_daemon/managed_daemon.go

```diff
@@ -0,0 +1,208 @@
+}
+
+func (md *ManagedDaemon) GetAgentCommunicationFileName() string {
+	return md.agentCommunicationFileName
+}
+
+func (md *ManagedDaemon) GetMountPoints() []*MountPoint {
+	return md.mountPoints
+}
+
+func (md *ManagedDaemon) GetEnvironment() map[string]string {
+	return md.environment
+}
+
+func (md *ManagedDaemon) SetMountPoints(mountPoints []*MountPoint) {
+	md.mountPoints = make([]*MountPoint, len(mountPoints))
+	copy(md.mountPoints, mountPoints)
+}
+
+func (md *ManagedDaemon) SetEnvironment(environment map[string]string) {
+	md.environment = make(map[string]string)
+	for key, val := range environment {
+		md.environment[key] = val
+	}
+}
+
+// AddMountPoint will add by MountPoint.SourceVolume
+// which is unique to the task and is a required field
+// and will throw an error if an existing
+// MountPoint.SourceVolume is found
+func (md *ManagedDaemon) AddMountPoint(mp *MountPoint) error {
+	mountIndex := md.GetMountPointIndex(mp)
+	if mountIndex != -1 {
+		return fmt.Errorf("MountPoint already exists at index %d", mountIndex)
+	}
+	md.mountPoints = append(md.mountPoints, mp)
+	return nil
+}
+
+// UpdateMountPoint will update by
+// MountPoint.SourceVolume which is unique to the task
+// and will throw an error if the MountPoint.SourceVolume
+// is not found
+func (md *ManagedDaemon) UpdateMountPointBySourceVolume(mp *MountPoint) error {
+	mountIndex := md.GetMountPointIndex(mp)
+	if mountIndex < 0 {
+		return fmt.Errorf("MountPoint %s not found; will not update", mp.SourceVolume)
+	}
+	md.mountPoints[mountIndex] = mp
+	return nil
+}
+
+// UpdateMountPoint will delete by
+// MountPoint.SourceVolume which is unique to the task
+// and will throw an error if the MountPoint.SourceVolume
+// is not found
+func (md *ManagedDaemon) DeleteMountPoint(mp *MountPoint) error {
+	mountIndex := md.GetMountPointIndex(mp)
+	if mountIndex < 0 {
+		return fmt.Errorf("MountPoint %s not found; will not delete", mp.SourceVolume)
+	}
+	md.mountPoints = append(md.mountPoints[:mountIndex], md.mountPoints[mountIndex+1:]...)
+	return nil
+}
+
+// GetMountPointIndex will return index of a mountpoint or -1
+// search by the unique MountPoint.SourceVolume field
+func (md *ManagedDaemon) GetMountPointIndex(mp *MountPoint) int {
+	sourceVolume := mp.SourceVolume
+	for i, mount := range md.mountPoints {
+		if mount.SourceVolume == sourceVolume {
+			return i
+		}
+	}
+	return -1
+}
+
+// AddEnvVar will add by envKey
+// and will throw an error if an existing
+// envKey is found
+func (md *ManagedDaemon) AddEnvVar(envKey string, envVal string) error {
+	_, exists := md.environment[envKey]
+	if !exists {
+		md.environment[envKey] = envVal
+		return nil
+	}
+	return fmt.Errorf("EnvKey: %s already exists; will not add EnvVal: %s", envKey, envVal)
+}
+
+// Updates environment varable by evnKey
+// and will throw an error if the envKey
+// is not found
+func (md *ManagedDaemon) UpdateEnvVar(envKey string, envVal string) error {
+	_, ok := md.environment[envKey]
```

Comment:
nit: could we use the same naming convention for this boolean? It switches between `exists` and `ok`

Reply:
That name was deliberate to differentiate between the expectation that it _shouldn't_ exist in one case and _should_ exist in the other.

Reply:
Gotcha, I guess it's up to personal preference of interpreting. Either way LGTM.